### PR TITLE
3889 k8s 1.22, 1.23 support

### PR DESCRIPTION
**Pull request by danielhoherd**
## Description

Add support for k8s 1.22, 1.23. Originally this was slated for 1.22, but 1.23 is out now so why not support it too, especially because even 1.22 will be EOL by 2023 for [upstream](https://github.com/kubernetes/website/blob/main/content/en/releases/patch-releases.md#122) and [AKS](https://docs.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar).

## Related Issues

- <https://github.com/astronomer/issues/issues/3889>
- Closes <https://github.com/astronomer/astronomer/issues/1370>

## Testing

If CI is passing that is one thing, but we should also make sure to test this with an airflow deployment before merging, so we can make sure that our airflow-chart works in 1.22+ when deployed through commander.

### CODE REVIEW

File: charts/keda/requirements.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 dependencies:
 - name: keda
-  version: 1.3.0
+  version: 2.6.2
```
Comment (pgvishnuram) [inline charts/keda/requirements.yaml:3]: are we trying to only put chart changes not images. whats the actual benefit out of it

Comment (danielhoherd) [reply charts/keda/requirements.yaml:3]: The old version had api incompatibilities.

Comment (vishwas-astro) [reply charts/keda/requirements.yaml:3]: Should we completely get rid of keda in our software?

Comment (danielhoherd) [reply charts/keda/requirements.yaml:3]: hah, i don't know. do we use it? do customers use it? i'm honestly not sure where we are at with this.

**Comment by danielhoherd** [comment]
I'm pretty sure this is good to be merged. @pgvishnuram do you know of anything else we need to do before merging this?

File: charts/keda/requirements.yaml (hunk 1 above)
Comment (danielhoherd) [reply charts/keda/requirements.yaml:3]: If we want to get rid of keda, we should do it in a different branch. I've opened an issue to track that discussion: <https://github.com/astronomer/issues/issues/4441>

**Comment by pgvishnuram** [comment]
Are we going to support 1.18 - if not we can merge this change

**Comment by danielhoherd** [comment]
@pgvishnuram I'd like to merge the removal of keda before merging this so that the keda upgrade included in this change set never makes it into the repo history. as for k8s 1.18, I'm not sure. More discussion on that point in <https://github.com/astronomer/issues/issues/3889#issuecomment-1081228761>

**Pull request merged**